libraries/AC_Autorotation/AC_Autorotation.cpp

```diff
@@ -384,3 +558,160 @@ float AC_Autorotation::calc_speed_forward(void)
+    } else {
+        _desired_sink_rate = 0.0f;
+    }
+    // update forward speed for logging
+    _speed_forward = calc_speed_forward(); //(cm/s)
+
+    _collective_out =  constrain_value((_p_coll_tch.get_p(_desired_sink_rate - _current_sink_rate))*0.01f + _ff_term_hs, 0.0f, 1.0f);
+    col_trim_lpf.set_cutoff_frequency(_col_cutoff_freq);
+    _ff_term_hs = col_trim_lpf.apply(_collective_out, _dt);
+    set_collective(HS_CONTROLLER_COLLECTIVE_CUTOFF_FREQ);
+    _pitch_target *= 0.95f;
+}
+
+void AC_Autorotation::get_entry_speed()
+{
+    _flare_entry_speed = calc_speed_forward();
+}
+
+void AC_Autorotation::time_to_ground()
+{
+    if (_inav.get_velocity_z_up_cms() < 0.0f ) {
+        _time_to_ground = -(_radar_alt/_inav.get_velocity_z_up_cms());
+    } else {
+        _time_to_ground = _t_tch +1.0f;
+    }
+}
+
+void AC_Autorotation::init_est_radar_alt()
+{
+    // set descent rate filter cutoff frequency
+    descent_rate_lpf.set_cutoff_frequency(40.0f);
+
+    // Reset feed descent rate filter
+    descent_rate_lpf.reset(_inav.get_velocity_z_up_cms());
+
+    _radar_alt_calc = _radar_alt;
+    _radar_alt_prev = _radar_alt;
+    _est_alt = _radar_alt;
+
+}
+
+void AC_Autorotation::update_est_radar_alt()
+{
+    if (_using_rfnd) {
+        // continue calculating radar altitude based on the most recent update and descent rate
+        if (is_equal(_radar_alt, _radar_alt_prev)) {
+            _radar_alt_calc += (_inav.get_velocity_z_up_cms() * _dt);
+        } else {
+            _radar_alt_calc = _radar_alt;
+            _radar_alt_prev = _radar_alt;
+        }
+        // determine the error between a calculated radar altitude based on each update at 20 hz and the estimated update
+        float alt_error = _radar_alt_calc - _est_alt;
+        // drive the estimated altitude to the actual altitude with a proportional altitude error feedback
+        float descent_rate_corr = _inav.get_velocity_z_up_cms() + alt_error * 2.0f;
+        // update descent rate filter
+        _descent_rate_filtered = descent_rate_lpf.apply(descent_rate_corr);
+        _est_alt += (_descent_rate_filtered * _dt);
+    } else {
+        _est_alt = _radar_alt;
+        // Reset feed descent rate filter
+        descent_rate_lpf.reset(_inav.get_velocity_z_up_cms());
+        // reset variables until using rangefinder
+        _radar_alt_calc = _radar_alt;
+        _radar_alt_prev = _radar_alt;
+        _est_alt = _radar_alt;
+    }
+}
+
+void AC_Autorotation::init_avg_acc_z()
+{
+    _avg_acc_z = 0.0f;
+    _acc_z_sum = 0.0f;
+    _index = 0;
+    memset(_curr_acc_z, 0, sizeof(_curr_acc_z));
+
+}
+
+void AC_Autorotation::calc_avg_acc_z()
+{
+    if (_index < 10) {
+        _acc_z_sum -= _curr_acc_z[_index];
+        _curr_acc_z[_index]= _ahrs.get_accel_ef().z;
+        _acc_z_sum += _curr_acc_z[_index];
+        _index = _index+1;
+    } else {
+        _index = 0;
+    }
```

Comment:
```suggestion
    if (_index >= 10) {
       _index = 0;
    }

    _acc_z_sum -= _curr_acc_z[_index];
    _curr_acc_z[_index]= _ahrs.get_accel_ef().z;
    _acc_z_sum += _curr_acc_z[_index];
    _index = _index+1;

```

Reply:
Were missing every 11th sample from the average.  The suggestion should fix it.

Reply:
done